### PR TITLE
fix/otel-jaeger: update jaeger deployment

### PR DESCRIPTION
**Pull request by DaedalusG**
### Checklist

- [ ] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [ ] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)
- [ ] Update [Kubernetes update doc](https://docs.sourcegraph.com/admin/updates/kubernetes)

### Test plan
Deploy locally and validate

### CODE REVIEW

**Review by bobheadxi — APPROVED** [review]

File: charts/sourcegraph/templates/jaeger/jaeger.Deployment.yaml [hunk 1]
```diff
@@ -53,7 +53,7 @@ spec:
       - name: jaeger
         image: {{ include "sourcegraph.image" (list . "jaeger") }}
         imagePullPolicy: {{ .Values.sourcegraph.image.pullPolicy }}
-        args: {{- default (list "--memory.max-traces=20000") .Values.jaeger.args | toYaml | nindent 8 }}
+        args: {{- default (list "--memory.max-traces=20000" "--sampling.strategies-file=/etc/jaeger/sampling_strategies.json" "--collector.otlp.enabled") .Values.jaeger.args | toYaml | nindent 8 }}
```
Comment (bobheadxi) [inline charts/sourcegraph/templates/jaeger/jaeger.Deployment.yaml:56]: Minor: does this mean if one provides their own args to the chart, they need to make sure to include the default args explicitly?

Comment (DaedalusG) [reply charts/sourcegraph/templates/jaeger/jaeger.Deployment.yaml:56]: Yeah it does, if they provide their own entry args these will be overwritten. Thanks for pointing it out, something for docs. Maybe worth it to add the args to the `values.yml` so they are recommended there as a default as well? cc: @michaellzc

**Review by michaellzc — APPROVED** [review]

**Review by Chickensoupwithrice — APPROVED** [review]

**Comment by Chickensoupwithrice** [comment]
Tested deployment locally with kind, and was successful after a couple fixes :)

File: charts/sourcegraph/templates/otel-collector/otel-collector.Deployment.yaml [hunk 2]
```diff
@@ -64,9 +64,9 @@ spec:
         - name: JAEGER_HOST
           value: jaeger-collector
         - name: JAEGER_OTLP_GRPC_PORT
-          value: 4320
+          value: "4320"
```
Comment (DaedalusG) [inline charts/sourcegraph/templates/otel-collector/otel-collector.Deployment.yaml:67]: dang, I wonder if I can get a linter that will detect this

**Pull request merged**